### PR TITLE
Renamed Task to PTask to resolve naming conflict

### LINKED ISSUE #288: Cannot build with Xcode13 / IOS 15
Multiple errors like the ones below result in a failed build.

ios/Pods/MultiplatformBleAdapter/iOS/RxBluetoothKit/RestoredState.swift:68:105: error: value of optional type 'CBPeripheral?' must be unwrapped to a value of type 'CBPeripheral'
ios/Pods/MultiplatformBleAdapter/iOS/RxBluetoothKit/RxCBCharacteristic.swift:59:52: error: value of optional type 'CBService?' must be unwrapped to a value of type 'CBService'
ios/Pods/MultiplatformBleAdapter/iOS/RxBluetoothKit/RxCBDescriptor.swift:43:62: error: value of optional type 'CBCharacteristic?' must be unwrapped to a value of type 'CBCharacteristic'

### PR DESCRIPTION
**Pull request by DevNico**
This fixes the build problems with Xcode 13. The main fix was to rename the `Task` to `PTask` as in `PeripheralTask` to resolve a naming conflict with new Swift features. Additionally, some `!` unwrapping of variables was added.

This is literally the first time I touched Swift code. I cannot guarantee **everything** works. But I tested this fix in our app and everything works as expected. In our use case we connect to a device and read/write two characteristics.

Closes #288.

### CODE REVIEW

File: ios/Classes/BleData extras/QualifiedCharacteristic.swift [hunk 1]
```diff
@@ -13,8 +13,8 @@ extension QualifiedCharacteristic {
     init(_ characteristic: CBCharacteristic) {
         self.init(
             id: characteristic.uuid,
-            serviceID: characteristic.service.uuid,
-            peripheralID: characteristic.service.peripheral.identifier
+            serviceID: characteristic.service!.uuid,
+            peripheralID: characteristic.service!.peripheral!.identifier
```
Comment (werediver) [inline ios/Classes/BleData extras/QualifiedCharacteristic.swift:17]: Optionality was matching before, but not anymore? Please, comment on every case (why exactly? why not guard-throw or init?-null)

File: ios/Classes/ReactiveBle/Tasks/PeripheralTaskRegistry/PeripheralTaskController.swift [hunk 2]
```diff
@@ -2,14 +2,14 @@ protocol PeripheralTaskController {
 
     associatedtype TaskSpec: PeripheralTaskSpec
 
-    typealias Task = PeripheralTask<TaskSpec>
+    typealias PTask = PeripheralTask<TaskSpec>
```
Comment (werediver) [inline ios/Classes/ReactiveBle/Tasks/PeripheralTaskRegistry/PeripheralTaskController.swift:5]: Single-letter prefix is too obscure. Consider using `TaskType` (not ideal, but), or `SubjectTask`, or something else that reads fine and makes sense. Please, use the chosen approach consistently throughout the rest of the code.

**Review by werediver — CHANGES_REQUESTED** [review]
Thanks for looking into this.

File: ios/Classes/ReactiveBle/Tasks/PeripheralTaskRegistry/PeripheralTaskController.swift (hunk 2 above)
Comment (DevNico) [reply ios/Classes/ReactiveBle/Tasks/PeripheralTaskRegistry/PeripheralTaskController.swift]: I renamed `PTask` to `SubjectTask` as you suggested.

File: ios/Classes/BleData extras/QualifiedCharacteristic.swift (hunk 1 above)
Comment (DevNico) [reply ios/Classes/BleData extras/QualifiedCharacteristic.swift:17]: As I mentioned Swift is a language I never really use. I unwrapped the variables with a `!` just to fix the problem. I tested it and it worked. I don't have enough Swift / native iOS development experience to properly solve the problem. This is something someone else would have to look into with more experience.

File: ios/Classes/ReactiveBle/Tasks/PeripheralTaskRegistry/PeripheralTaskRegistry.swift [hunk 3]
```diff
@@ -2,21 +2,21 @@ import Foundation
 
 final class PeripheralTaskRegistry<Controller: PeripheralTaskController> {
 
-    typealias Task = Controller.Task
-    typealias TaskCompletionHandler = (Task.Key, Task.Params, Task.Result) -> Void
+    typealias CTask = Controller.SubjectTask
```
Comment (werediver) [inline ios/Classes/ReactiveBle/Tasks/PeripheralTaskRegistry/PeripheralTaskRegistry.swift:5]: `CTask` is no better than `PTask` was, so I suggest using that `SubjectTask` name here as well. Or I can take it along with the force-unwrapping question if/when I have time to.

File: ios/Classes/BleData extras/QualifiedCharacteristic.swift (hunk 1 above)
Comment (werediver) [reply ios/Classes/BleData extras/QualifiedCharacteristic.swift:17]: Clear. I appreciate your effort any way, but this is something to be clarified and refined before we merge this p/r.

I'll try to find time to look into this in the coming week or two.

**Review by werediver — CHANGES_REQUESTED** [review]
One step closer, but still has questions to be answered. As I mentioned in one of the comments, I'll try to find time to look into this in the coming week or two.

Thank you for your effort.

**Comment by DevNico** [comment]
Important note:

Building with XCode 12 will not work anymore with the `!` unwrapping changes. Just FYI

**Comment by SamiKouatli** [comment]
Thank you, it was what I was looking for !

**Comment by remonh87** [comment]
@DevNico thank you so much for your work. Due to the big repo changes I need to close this and  I created #365 to incorporate your changes. I will make sure to give you credits in the release notes.

**Pull request closed without merging**